src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -83,7 +99,7 @@ def get(self, request: Request, project) -> Response:
                 scope.set_tag("integration_provider", provider)
                 scope.set_tag("stacktrace_link.platform", platform)
 
-                if not filepath.startswith(config.stack_root):
+                if not filepath.startswith(config.stack_root) and not frame:
                     scope.set_tag("stacktrace_link.error", "stack_root_mismatch")
                     result["error"] = "stack_root_mismatch"
                     continue
```

Comment:
I feel like we might still want to set this error if we fail to retrieve frames while stack munging?

Reply:
Do you mean we should return that same error if the original `file` does not return a repo-link and the `munged_filename` also doesn't return a repo-link?

Reply:
I'm not totally sure, but the main concern I have is that if you pass frame info then we'll always skip the if statement above. It might be fine, probably just worth getting feedback from ecosystem.

Reply:
Hmm, I see what you mean. 

I tested the scenario when the params are present but the `file` does not resolve to a repo-link and the only difference is the `result["error"]` value is `file_not_found` and not `stack_root_mismatch`. Not sure if there's a difference in these two errors. 

Maybe @getsentry/ecosystem can advise.